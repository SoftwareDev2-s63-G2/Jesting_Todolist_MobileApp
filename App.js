import * as React from 'react';
import { View, Text } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { navigationRef } from './RootNavigation';
import Home from './screens/Home'
import EditingTodo from './screens/EditingTodo'
import AddingTodo from './screens/AddingTodo'

const Stack = createNativeStackNavigator();

function App() {
  return (
    <NavigationContainer ref={navigationRef}>
      <Stack.Navigator screenOptions={{ headerShown: false }}>
        <Stack.Screen name="Home" component={Home} />
        <Stack.Screen name="Add"  component={AddingTodo} />
        <Stack.Screen name="Edit" component={EditingTodo} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default App;