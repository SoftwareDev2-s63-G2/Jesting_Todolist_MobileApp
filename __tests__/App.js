jest.useFakeTimers();
import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
 
describe("<App />", () => {
    it("has children" , () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree.children.length).toBeGreaterThanOrEqual(1);
    });
   it('renders correctly', () => {
       const tree = renderer.create(<App />).toJSON();
       expect(tree).toMatchSnapshot();
   });
});
