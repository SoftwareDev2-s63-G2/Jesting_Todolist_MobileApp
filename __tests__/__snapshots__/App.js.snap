// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#222831",
              "color": "#EEEEEE",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00ADB57D",
                "color": "#EEEEEE",
                "flexDirection": "row",
                "height": 120,
                "justifyContent": "center",
                "paddingHorizontal": 10,
                "paddingTop": 26,
                "width": "100%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Search ..."
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#EEEEEE",
                  "borderRadius": 24,
                  "color": "black",
                  "fontSize": 15,
                  "height": 40,
                  "marginRight": 15,
                  "paddingHorizontal": 10,
                  "width": "70%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 50,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={30}
                bbWidth={30}
                fill="#EEEEEE"
                focusable={false}
                height={30}
                meetOrSlice={0}
                minX={0}
                minY={0}
                secondaryFill={null}
                secondaryOpacity={0.4}
                size={30}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {},
                    Object {
                      "flex": 0,
                      "height": 30,
                      "width": 30,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={512}
                width={30}
              >
                <RNSVGGroup
                  fill={4293848814}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
                    fill={4293848814}
                    fillOpacity={1}
                    propList={
                      Array [
                        "fill",
                        "fillOpacity",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 50,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={30}
                bbWidth={30}
                fill="#EEEEEE"
                focusable={false}
                height={30}
                meetOrSlice={0}
                minX={0}
                minY={0}
                secondaryFill={null}
                secondaryOpacity={0.4}
                size={30}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {},
                    Object {
                      "flex": 0,
                      "height": 30,
                      "width": 30,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={448}
                width={30}
              >
                <RNSVGGroup
                  fill={4293848814}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
                    fill={4293848814}
                    fillOpacity={1}
                    propList={
                      Array [
                        "fill",
                        "fillOpacity",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#00ADB5",
                    "fontSize": 18,
                    "textAlignVertical": "top",
                  }
                }
              >
                No result found
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
